### ISSUE TITLE
LibJS: Hitting a VERIFY_NOT_REACHED() in Intl/NumberFormat.cpp on live website

### ISSUE THREAD

**Issue by awesomekling** [SerenityOS/serenity#20944]
When loading https://www.hemnet.se/ I consistently hit this thing at `IntlFormat.cpp:916`

```cpp
// viii. Else,
else {
    // 1. Let unknown be an ILND String based on x and p.
    // 2. Append a new Record { [[Type]]: "unknown", [[Value]]: unknown } as the last element of result.

    // LibUnicode doesn't generate any "unknown" patterns.
    VERIFY_NOT_REACHED();
}
```

cc @trflynn89

**Comment by trflynn89** [comment]
This crash is fixed in #20948, but then we crash later in streams:
```
#0  ak_verification_failed () at AK/Assertions.cpp:72
#1  0x00007ffff6aaf26d in AK::Optional<AK::Variant<JS::NonnullGCPtr<Web::Streams::ReadableStreamDefaultController>, JS::NonnullGCPtr<Web::Streams::ReadableByteStreamController> > >::value() & () at AK/Optional.h:215
#2  AK::Optional<AK::Variant<JS::NonnullGCPtr<Web::Streams::ReadableStreamDefaultController>, JS::NonnullGCPtr<Web::Streams::ReadableByteStreamController> > >::operator-> () at AK/Optional.h:289
#3  Web::Streams::readable_stream_default_reader_read () at Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp:483
#4  0x00007ffff6ac82c8 in Web::Streams::ReadableStreamDefaultReader::read () at Userland/Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp:182
#5  0x00007ffff704397d in operator() () at Build/lagom/Userland/Libraries/LibWeb/Bindings/ReadableStreamDefaultReaderPrototype.cpp:143
#6  throw_dom_exception_if_needed<Web::Bindings::ReadableStreamDefaultReaderPrototype::read(JS::VM&)::<lambda()> > () at Userland/Libraries/LibWeb/Bindings/ExceptionOrUtils.h:97
#7  Web::Bindings::ReadableStreamDefaultReaderPrototype::read () at Build/lagom/Userland/Libraries/LibWeb/Bindings/ReadableStreamDefaultReaderPrototype.cpp:143
#8  0x00005555555da2b5 in JS::SafeFunction<JS::ThrowCompletionOr<JS::Value> (JS::VM&)>::CallableWrapper<JS::ThrowCompletionOr<JS::Value> (*)(JS::VM&)>::call(JS::VM&) () at Userland/Libraries/LibJS/SafeFunction.h:133
#9  0x00007ffff5d8331a in JS::SafeFunction<JS::ThrowCompletionOr<JS::Value> (JS::VM&)>::operator()(JS::VM&) const () at Userland/Libraries/LibJS/SafeFunction.h:85
#10 JS::NativeFunction::call () at Userland/Libraries/LibJS/Runtime/NativeFunction.cpp:222

```

**Comment by awesomekling** [comment]
Thanks, Tim!
